biz.aQute.repository/test/aQute/bnd/repository/fileset/FileSetRepositoryTest.java

```diff
@@ -0,0 +1,51 @@
+package aQute.bnd.repository.fileset;
+
+import static aQute.lib.exceptions.FunctionWithException.asFunction;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+
+public class FileSetRepositoryTest {
+
+	@Test
+	public void includesMavenArtifacts() throws Exception {
+		List<File> files = classpath();
+
+		assertTrue(files.size() > 0);
+
+		FileSetRepository repository = new FileSetRepository("test", files);
+
+		List<String> list = repository.list(null);
+
+		assertTrue(list.contains("biz.aQute.http.testservers"));
+		assertFalse(list.contains("javax.annotation:jsr250-api"));
```

Comment:
Can't we just use jars from the `testdata` or `testresources` folders rather than add some to the testpath just for this test method? For example, use `testresources/mavenrepo/commons-cli/commons-cli/1.0/commons-cli-1.0.jar` as a jar without a BSN or pom xml resource. It seems contrived for the purposes of testing FileSetRepository that we must use the actual class path of the test case. It is too risky should we change the build later. The testpath should be about test execution and not testdata.

Reply:
I really hate adding binaries for test data.

Reply:
.. but fine.

Reply:
The alternate is to create a MavenBndRepository object, like we do in some tests, and have it download the binaries you need for testing. But adding jars to testpath when the test cases do not need to compile or run against them is wrong.